sycl-fusion/jit-compiler/include/KernelFusion.h

```diff
@@ -54,16 +54,28 @@ class FusionResult {
 class KernelFusion {
 
 public:
-  static FusionResult fuseKernels(
-      Config &&JITConfig, const std::vector<SYCLKernelInfo> &KernelInformation,
-      const std::vector<std::string> &KernelsToFuse,
-      const std::string &FusedKernelName,
-      jit_compiler::ParamIdentList &Identities, BarrierFlags BarriersFlags,
-      const std::vector<jit_compiler::ParameterInternalization>
-          &Internalization,
-      const std::vector<jit_compiler::JITConstant> &JITConstants);
+  static FusionResult
+  fuseKernels(const std::vector<SYCLKernelInfo> &KernelInformation,
+              const std::vector<std::string> &KernelsToFuse,
+              const std::string &FusedKernelName,
+              jit_compiler::ParamIdentList &Identities,
+              BarrierFlags BarriersFlags,
+              const std::vector<jit_compiler::ParameterInternalization>
+                  &Internalization,
+              const std::vector<jit_compiler::JITConstant> &JITConstants);
 };
 
+/// Clear all previously set options.
+void resetConfiguration();
+
+/// Take ownership of \p Option and include it in the current configuration.
+void set(OptionPtrBase *Option);
+
+/// Set \p Opt to \p Value in the current configuration.
+template <typename Opt> void set(const typename Opt::ValueType &Value) {
+  set(new Opt{Value});
+}
```

Comment:
```suggestion
/// Set \p Opt to the value built by \p As, built in place.
template <typename Opt, typename... Args> 
void set(Args&&... As) {
  set(new Opt{std::forward<Args>(As)...});
}
```
Allow users to build option in place (goes only with change proposed below)